### DISCUSSION TITLE
Feature request: Time based charging cost

### DISCUSSION THREAD

**Discussion by tfsjohan** [teslamate-org/teslamate#1747] [Ideas]
I'd like if it was possible to set different charging costs basen on time of day and dates. I mostly charge at home and I have solar panels. During the spring and summer I mainly, but not always, charge my car only when the solar panels generates enough to supply both the house and the car with energy. So this energy is free, but if I charge at night it will still cost me. Also, I can only do thing from like April to September. 

It would be really nice if a geolocation could add criteria for charing costs, both for calendar date and time of day with in a date span.

**Comment by Dulanic** [comment]
I thought of this myself, but I just found that I always charge at night because my cost is lowest. This seems like a very specific use case, I'd just edit them manually if I was you once you know it was free? Easy enough to edit the charge cost under Charges dashboard.

**Comment by adriankumpf** [comment]
Feature requests belong here: [💡 Ideas](https://github.com/adriankumpf/teslamate/discussions/categories/ideas). I'll move it.
